packages/react/src/time-picker/TimePicker.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react';
+import { useCombobox } from 'downshift';
+import { type ComboboxProps } from '../combobox/Combobox';
+import { ComboboxBase } from '../combobox/ComboboxBase/ComboboxBase';
+import {
+	type DefaultComboboxOption,
+	useComboboxInputId,
+} from '../combobox/utils';
+import { type TimeFormat } from '../time-input/utils';
+import { filterOptions, generateOptions } from './utils';
+
+export function TimePicker({
+	id,
+	inputRef: inputRefProp,
+	interval = 15,
+	loading = false,
+	max = '24:00',
+	maxWidth = 'md',
+	min = '00:00',
+	onChange,
+	timeFormat = 'h:mm aaa',
+	value,
+	...props
+}: TimePickerProps) {
+	const {
+		// We remove this prop so that it can't be passed to ComboboxBase
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		renderItem,
+		...consumerProps
+	} = props;
+	const inputId = useComboboxInputId(id);
+	const [options, setOptions] = useState<DefaultComboboxOption[]>(
```

Comment:
You should be able to remove the `<DefaultComboboxOption[]>` as the type is inferred from `generateOptions`.

As a general rule with `useState` and `useRef`, if you pass in an initial value that you intend to be the only type accepted, then you don't need to define a type in the `< >`.

Reply:
Not sure which is better. Removing it results in 
<img width="581" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/1df7c64c-bd76-4f85-8d81-1bc836049ebf">
Adding it results in 
<img width="738" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/7b22410c-c272-4c7c-b374-fded24fdc4df">
<img width="551" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/3b9ea883-d88d-478b-9876-e1a8a8d60f66">

Reply:
What do you get if you hover over `options`?